modules/optimization/src/executioners/OptimizeSolve.C

```diff
@@ -212,7 +212,9 @@ OptimizeSolve::taoSolve()
 
   // save nonTAO PETSC options to reset before every call to execute()
   _petsc_options = _problem.getPetscOptions();
-  _solver_params = _problem.solverParams();
+  // We're logically asssociated with the zeroth solver system because our inner solve is a
+  // FixedPointSolve whose inner solve is a FEProblemSolve whose solver system is hard-coded to 0
```

Comment:
FEProblemSolve can do multi-system

I think it s more OptimizeSolve that is hardcoded to handle only one